src/gs-application.c

```diff
@@ -260,7 +260,7 @@ theme_changed (GtkSettings *settings, GParamSpec *pspec, GsApplication *app)
 static void
 gs_application_shell_loaded_cb (GsShell *shell, GsApplication *app)
 {
-	gs_shell_set_mode (app->shell, GS_SHELL_MODE_OVERVIEW);
+	g_signal_handler_disconnect (app->shell, app->shell_loaded_handler_id);
```

Comment:
Is this whole commit really needed? Is the problem that `gs_application_shell_loaded_cb` is getting called before the `initial_overview_load_done` is called?

Reply:
It's not _needed_, no - I just think it's confusing to split the state machine between application and shell, unless application is doing something different to the default.